unleash-yggdrasil/src/lib.rs

```diff
@@ -303,9 +313,39 @@ impl EngineState {
         }
     }
 
+    fn is_parent_dependency_satisfied(&self, toggle: &CompiledToggle, context: &Context) -> bool {
+        if toggle.dependencies.is_empty() {
+            return true;
+        }
+
+        toggle.dependencies.iter().all(|parent| {
+            let Some(parent_toggle) = self.get_toggle(&parent.feature) else {
+                return false;
+            };
+
+            if !parent_toggle.dependencies.is_empty() {
+                return false;
+            }
+
+            if parent.enabled.unwrap_or(true) {
+                match parent.variants.as_ref() {
+                    Some(variants) if !variants.is_empty() => {
+                        variants.contains(&self.get_variant(&parent.feature, context).name)
```

Comment:
This looks a little off to me. If we have variants we never check the parent's actual state, or did I miss something?

Reply:
Good question! I'll ... check. But that sounds like something the spec tests should have caught. I'll see what I can find.

Reply:
Yes, completely agree, the spec tests should have caught this but looking it over I don't think we have a case where we check both the constraint and a variant

Reply:
Wait, hang on; I think that checks out. Because we check whether parent flag's returned variant is one of the variants listed in the dependencies. If the parent flag is disabled, then the returned variant is the disabled variant, right? So implicitly, this assumes that it is.

Now, that *does* maybe open up the door for one edge case:
If the child depends on the parent being enabled and yielding the "disabled" variant. 
In that case, the dependency will be satisfied even if the parent flag isn't enabled 🤔

Reply:
But yeah, you're right! Added that as a test case to the spec in https://github.com/Unleash/client-specification/pull/65 (and locally) and I've fixed that point. Thanks 😄